viewer/viewer.js

```diff
@@ -7686,24 +7686,126 @@ function buildHuntOptions (huntId, hunt) {
   return options;
 }
 
+// if we couldn't retrieve the seession, skip it but add it to failedSessionIds
+// so that we can go back and search for it at the end of the hunt
+function continueHuntSkipSession (hunt, huntId, session, sessionId, searchedSessions, cb) {
+  if (!hunt.failedSessionIds) {
+    hunt.failedSessionIds = [ sessionId ];
+  } else {
+    // pause the hunt if there are more than 10k failed sessions
+    if (hunt.failedSessionIds.length > 10000) {
+      return pauseHuntJobWithError(huntId, hunt, {
+        value: 'Error hunting: Too many sessions are unreachable. Please contact your administrator.'
+      });
+    }
+    // make sure the session id is not already in the array
+    // if it's not the first pass and a node is still down, this could be a duplicate
+    if (hunt.failedSessionIds.indexOf(sessionId) < 0) {
+      hunt.failedSessionIds.push(sessionId);
+    }
+  }
+
+  return updateHuntStats(hunt, huntId, session, searchedSessions, cb);
+}
+
+// if there are failed sessions, go through them one by one and do a packet search
+// if there are no failed sessions left at the end then the hunt is done
+// if there are still failed sessions, but some sessions were searched during the last pass, try again
+// if there are still failed sessions, but no new sessions coudl be searched, pause the job with an error
+function huntFailedSessions (hunt, huntId, options, searchedSessions, user) {
+  let changesSearchingFailedSessions = false;
+  if (hunt.failedSessionIds && hunt.failedSessionIds.length) {
```

Comment:
style thing, would prefer a ! with a return here, so the rest of the function isn't indented.

Reply:
👍